applications/serial_lte_modem/src/slm_at_fota.c

```diff
@@ -375,15 +443,67 @@ void slm_fota_post_process(void)
 
 void slm_finish_modem_fota(int modem_lib_init_ret)
 {
-	if (handle_nrf_modem_lib_init_ret(modem_lib_init_ret)) {
-		char buf[40];
+	char buf[40];
+#if defined(CONFIG_SLM_FULL_FOTA)
+	int err;
+
+	ARG_UNUSED(modem_lib_init_ret);
+
+	if (fota_type == DFU_TARGET_IMAGE_TYPE_FULL_MODEM) {
+		/* Full fota activation differs from delta modem fota. */
+		LOG_INF("Applying full modem firmware update from external flash\n");
+
+		err = nrf_modem_lib_shutdown();
```

Comment:
How do we recover if any of the operations in here fail? Have these failures been tested?

Reply:
According to my knowledge, the errors in this phase cannot be recovered but the downloaded firmware has to be ignored